homeassistant/components/google_generative_ai_conversation/entity.py

```diff
@@ -301,14 +302,13 @@ async def _transform_stream(
 class GoogleGenerativeAILLMBaseEntity(Entity):
     """Google Generative AI base entity."""
 
-    _attr_has_entity_name = True
-    _attr_name = None
-
-    def __init__(self, entry: ConfigEntry) -> None:
+    def __init__(self, entry: ConfigEntry, subentry: ConfigSubentry) -> None:
         """Initialize the agent."""
         self.entry = entry
+        self.subentry = subentry
+        self._attr_name = subentry.title or DEFAULT_CONVERSATION_NAME
```

Comment:
The default should be based on subentry type.

Reply:
I expect sub entry renaming to become possible in future versions which would this guard for. I'm fine removing it.